products/bridge/smart-contracts/contracts/Relayer.sol

```diff
@@ -107,9 +110,46 @@ contract Relayer {
             call,
             readonly,
             callback,
-            nonces[msg.sender]
+            _nonces[msg.sender]
         );
-        return ++nonces[msg.sender];
+        return ++_nonces[msg.sender];
+    }
+
+    // TODO: remove later, used for testing
+    function warmup() external {
+        ++_refund[msg.sender];
+    }
+
+    function topUpGas(address target) external payable {
+        _gas[target] += msg.value;
+    }
+
+    function refundGas() external {
+        uint amount = _refund[msg.sender];
+        _refund[msg.sender] = 0;
+        payable(msg.sender).transfer(amount);
+    }
+
+    function refund(address x) external view returns (uint) {
+        return _refund[x];
+    }
+
+    function gas(address x) external view {
```

Comment:
This compiles but doesn't return anything :-)
Add `returns (uint)` here and `return` before `_gas[x];` in the next line.
Alternatively, you can declare `_gas` and `_refund` as `public` and the compiler auto-generates the getter methods so you can remove them from the code.

Reply:
I was aware of that, was just not sure if making it public would mean that it can be modified externally. I am pretty sure the compiler does not auto-generate setters, just wanted to be sure tho.